### PR TITLE
Quote uppercase object names on Oracle

### PR DESCRIPTION
**Pull request by MarcinHoppe**

### CODE REVIEW

**Comment by MarcinHoppe** [comment]
@DavidBoike @SimonCropp @SzymonPobiega An attempt to straighten the quoting and capsing of identifiers on Oracle. Please review and merge if you think this looks good.

**Comment by DavidBoike** [comment]
Honestly can't believe it's only 20 files. Waiting for something to blow up. :wink:

File: src/ScriptBuilder.Tests/Saga/SagaScriptBuilderTest.CreateWithCorrelation.ForScenario.Oracle.approved.txt [hunk 1]
```diff
@@ -71,7 +71,7 @@ end if;
 select count(*) into n from user_indexes where table_name = 'THESAGA' and index_name = 'SAGAIDX_E78AFF37B286FF56612DF6';
 if(n = 0)
 then
-  sqlStatement := 'create unique index SAGAIDX_E78AFF37B286FF56612DF6 on THESAGA (CORR_CORRELATIONPROPERTY ASC)';
+  sqlStatement := 'create unique index "SAGAIDX_E78AFF37B286FF56612DF6" on "THESAGA" (CORR_CORRELATIONPROPERTY ASC)';
```
Comment (DavidBoike) [inline src/ScriptBuilder.Tests/Saga/SagaScriptBuilderTest.CreateWithCorrelation.ForScenario.Oracle.approved.txt]: Is there any reason for column names based on saga properties to also be quoted?

Comment (MarcinHoppe) [inline src/ScriptBuilder.Tests/Saga/SagaScriptBuilderTest.CreateWithCorrelation.ForScenario.Oracle.approved.txt]: You're thinking C# identifiers (property names) with funky characters?

**Comment by MarcinHoppe** [comment]
Green!

**Comment by DavidBoike** [comment]
Added a commit that uses names with spaces for the tests in Oracle. By that, I verified that you didn't miss anything @MarcinHoppe, although I did detect a place where a rebase mixup caused GetTablePrefix() not to be used in a test, and corrected that.

For sagas, added a test for a table suffix containing a space that passes for all engines.

Is this the way to go? Perhaps we should keep the saga test, but change the other persistence types to just have a single test that tests spaces on all engines?

**Pull request merged**